src/overlays/actors/ovl_En_Test6/z_en_test6.c

```diff
@@ -35,50 +53,1278 @@ const ActorInit En_Test6_InitVars = {
+            EnTest6_SetupAction(this, func_80A92950);
+            break;
+    }
+}
+
+void func_80A91690(EnTest6* this, GlobalContext* globalCtx) {
+    this->unk_274 = 90;
+    this->unk_27A = 100;
+    this->unk_286 = 0;
+    if (ENTEST6_GET(&this->actor) == ENTEST6_25) {
+        play_sound(NA_SE_SY_TIME_CONTROL_SLOW);
+    } else if (ENTEST6_GET(&this->actor) == ENTEST6_24) {
+        play_sound(NA_SE_SY_TIME_CONTROL_NORMAL);
+    }
+}
+
+void func_80A916F0(EnTest6* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->actor.freezeTimer = 0;
+    globalCtx->unk_18844 = 0;
+    ActorCutscene_Stop(globalCtx->playerActorCsIds[8]);
+    func_800B7298(globalCtx, NULL, 6);
+    func_80A90C34();
+    Distortion_ClearType(0x20);
+    Actor_MarkForDeath(&this->actor);
+}
+
+void func_80A91760(EnTest6* this, GlobalContext* globalCtx) {
+    Input* input = CONTROLLER1(globalCtx);
+    s16 temp_s0;
+    Player* player = GET_PLAYER(globalCtx);
+    Camera* sp78;
+    Vec3f sp6C;
+    Vec3f sp60;
+    Vec3f sp54;
+    s32 i;
+    f32 sp4C;
+    Camera* temp_s3 = Play_GetCamera(globalCtx, this->unk_284);
+
+    sp78 = Play_GetCamera(globalCtx, CAM_ID_MAIN);
+
+    switch (this->unk_274) {
+        case 90:
+            this->unk_276 = 2;
+            this->unk_15C = 0.0f;
+            this->unk_14C = 0.1f;
+            this->unk_282 = 0;
+            this->unk_278 = 0;
+            this->unk_274 = 91;
+            break;
+
+        case 91:
+            this->unk_15C += this->unk_14C;
+            func_800FD59C(globalCtx, &D_80A9404C, this->unk_15C);
+            func_800FD5E0(globalCtx, &D_80A94050, this->unk_15C);
+            func_800FD654(globalCtx, &D_80A94048, this->unk_15C);
+            func_800FD698(globalCtx, D_80A94054, D_80A94058, this->unk_15C);
+
+            if (this->unk_27A == 90) {
+                this->unk_282 = 0;
+                if (ENTEST6_GET(&this->actor) == ENTEST6_24) {
+                    this->unk_27C = 0x200;
+                    this->unk_150 = 0.0f;
+                    sp4C = -100.0f;
+                } else {
+                    this->unk_27C = 0x570;
+                    this->unk_150 = 110.0f;
+                    sp4C = 100.0f;
+                }
+                this->unk_14C = 1.0f;
+
+                for (i = 0; i < ARRAY_COUNT(this->unk_20C); i++) {
+                    this->unk_20C[i].x = player->actor.world.pos.x;
+                    this->unk_20C[i].y = player->actor.world.pos.y;
+                    this->unk_20C[i].z = player->actor.world.pos.z;
+                }
+
+                this->unk_254 = ZeldaArena_Malloc(sizeof(Vec3f) * 64);
+                if (this->unk_254 != NULL) {
+                    for (i = 0; i < ARRAY_COUNT(this->unk_254[0]); i++) {
+                        (*this->unk_254)[i].x = (((2.0f * Rand_ZeroOne()) - 1.0f) * 40.0f) + temp_s3->eye.x +
+                                                ((temp_s3->at.x - temp_s3->eye.x) * .2f);
```

Comment:
Is the lack of `0` on this float required to match?
```suggestion
                        (*this->unk_254)[i].x = (((2.0f * Rand_ZeroOne()) - 1.0f) * 40.0f) + temp_s3->eye.x +
                                                ((temp_s3->at.x - temp_s3->eye.x) * 0.2f);
```